src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -845,6 +845,15 @@ export class NotebookModel extends Disposable implements INotebookModel {
 			if (newConnection) {
 				if (newConnection.serverCapabilities?.notebookKernelAlias) {
 					this._currentKernelAlias = newConnection.serverCapabilities.notebookKernelAlias;
+					// Removes SQL kernel to Kernel Alias Connection Provider map
+					let sqlConnectionProvider = this._kernelDisplayNameToConnectionProviderIds.get('SQL');
+					if (sqlConnectionProvider) {
+						let index = sqlConnectionProvider.indexOf(newConnection.serverCapabilities.notebookKernelAlias.toUpperCase());
```

Comment:
is the kernel alias guaranteed to be uppercase?

Reply:
i.e. do we do this translation somewhere else?

Reply:
The connection Providers will only contain UpperCase connection providers ('KUSTO' for example), therefore indexOf is case sensitive and notebookKernelAlias would be of form 'Kusto'. Without the uppercase, it would not get the index properly.

Reply:
What enforces that it's uppercase, though?

If I go and create a new extension that contributes a notebook alias and the alias is 'chris' in package.json, do we do a `toUpperCase()` somewhere when looking for alias registrations?

Reply:
@VasuBhog Where is that guaranteed? I don't remember seeing us doing any upper casing of provider IDs - Kusto just defines its ID as upper case for no apparent reason that I know of : 

`https://github.com/microsoft/azuredatastudio/blob/main/extensions/kusto/package.json#L216`

Reply:
@Charles-Gagnon and @chlafreniere. I was going based on how the connection providers are stated both in the package.json for MSSQL and KUSTO, and how they appear in the mapping for connectionProviderIds. They are not cast to upper case but are defined that way. As a result, I wanted to ensure that indexOf gets the index properly for kernel alias providers if we continue to follow the upper case provider id format.

Reply:
Yes, but I'm saying that's just a purely coincidental thing currently. There's nothing preventing someone from defining a provider with a lowercase ID that I know of and so we should never be assuming that they're all upper case. 

That being said we should have a better solution for this. Doing something like upper casing all provider IDs no matter how they're defined in the package.json (or at least always doing case-insensitive checks) isn't unreasonable but we should make that part of the actual core logic first before writing code like this that assumes things that aren't guaranteed to be true.

Reply:
@Charles-Gagnon noted. I will be sure to address this, casting all provider IDs to the upper case prior to the indexOf ensures that future cases of lowercase provider IDs are also accounted for. Thanks for the feedback.